pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -682,8 +620,6 @@ def test_maybe_promote_datetimetz_with_any_numpy_dtype(
     )
 
 
-# override parametrization due to to many xfails; see GH 23982 / 25425
-@pytest.mark.parametrize("box", [(True, None), (True, object)])
```

Comment:
@jbrockmendel 
here we something fell through the cracks in #23982 - this tests never ran with `box=False`.

Reply:
it sounds like some of the changes in this test file are valid independent of the changes in the other file.  is that correct?

Reply:
Nope, all removed xfails are due to diverting the array-path through `maybe_promote_with_array` instead of `maybe_promote`. For `test_maybe_promote_datetimetz_with_any_numpy_dtype` and `test_maybe_promote_datetimetz_with_datetimetz` however, I needed to *add* xfails for the `box=False` case because those are not working within `maybe_promote` yet.